Content/NPCs/Vitric/Coolmite.cs

```diff
@@ -0,0 +1,277 @@
+﻿using StarlightRiver.Content.Bosses.GlassMiniboss;
+using StarlightRiver.Content.Dusts;
+using StarlightRiver.Content.Items.Vitric;
+using System;
+using System.IO;
+using System.Linq;
+using Terraria;
+using Terraria.Audio;
+using Terraria.DataStructures;
+using Terraria.GameContent.Bestiary;
+using Terraria.ID;
+using static Terraria.ModLoader.ModContent;
+
+namespace StarlightRiver.Content.NPCs.Vitric
+{
+	internal class Coolmite : ModNPC
+	{
+		bool melting = false;
+		int meltingTimer = 0;
+
+		public ref float ActionState => ref NPC.ai[0];
+		public ref float ActionTimer => ref NPC.ai[1];
+		public ref float GlobalTimer => ref NPC.ai[2];
+		public ref float TurnTimer => ref NPC.ai[3];
+		public float meltingTransparency => (float)meltingTimer / 150;
+
+		public override string Texture => "StarlightRiver/Assets/NPCs/Vitric/Coolmite";
+
+		public override void SetStaticDefaults()
+		{
+			DisplayName.SetDefault("Coolmite");
+			Main.npcCatchable[Type] = true;
+		}
+
+		public override void SetDefaults()
+		{
+			NPC.catchItem = ItemType<CoolmiteItem>();
+			NPC.width = 24;
+			NPC.height = 24;
+			NPC.damage = 0;
+			NPC.defense = 0;
+			NPC.lifeMax = 25;
+			NPC.aiStyle = -1;
+			NPC.lavaImmune = true;
+			NPC.HitSound = SoundID.Item27;
+
+			ActionState = -1;
+		}
+
+		public override void SetBestiary(BestiaryDatabase database, BestiaryEntry bestiaryEntry)
+		{
+			bestiaryEntry.Info.AddRange(new IBestiaryInfoElement[]
+			{
+				Bestiary.SLRSpawnConditions.VitricDesert,
+				new FlavorTextBestiaryInfoElement("[PH] Entry")
+			});
+		}
+
+		public override void SendExtraAI(BinaryWriter writer)
+		{
+			writer.WritePackedVector2(NPC.velocity);
+			writer.Write(melting);
+			writer.Write(meltingTimer);
+		}
+
+		public override void ReceiveExtraAI(BinaryReader reader)
+		{
+			NPC.velocity = reader.ReadPackedVector2();
+			melting = reader.ReadBoolean();
+			meltingTimer = reader.Read();
+		}
+
+		public override void AI()
```

Comment:
With the amount of code duplication here I wonder if it would be more sensible to have the coolmite inherit from the magmite, and then move the behavior of the magmites particle spawning to PostAI or similar?

Reply:
We may need to abstract the AI a bit to do so due to the lava targeting logic, though it likely wont be hard to abstract it to target a point